concent_api/common/decorators.py

```diff
@@ -54,13 +54,38 @@ def wrapper(*args: Any, **kwargs: Any) -> None:
     return wrapper
 
 
-def non_nesting_atomic(using: Union[str, Callable], savepoint: bool=True) -> Atomic:
-    if (
-        settings.DETECT_NESTED_TRANSACTIONS and
-        get_connection(using).in_atomic_block
-    ):
-        raise ConcentPendingTransactionError
-    if callable(using):
-        return Atomic(DEFAULT_DB_ALIAS, savepoint)(using)
-    else:
-        return Atomic(using, savepoint)
+class non_nesting_atomic(ContextDecorator):
```

Comment:
If it is class shouldn't name be camel-case?

Reply:
It could be but actually none of our tools is complaining (lint, flake, pycharm). I also think that when using this is more natural to have a decorator in this form than in camel case. It would look weirder if I do 

```
@NonNestingAtomic(using='control')
def something():
```

so I can sacrifice class name in one place in favor of using it as decorator in many places.